src/Arcus.WebApi.Security/Authentication/CertificateAuthenticationConfig.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Arcus.WebApi.Security.Authentication.Interfaces;
+using GuardNet;
+using Microsoft.Extensions.Logging;
+
+namespace Arcus.WebApi.Security.Authentication
+{
+    /// <summary>
+    /// Representation of the configurable validation requirements on a <see cref="X509Certificate2"/>.
+    /// </summary>
+    public class CertificateAuthenticationConfig
+    {
+        private readonly IDictionary<X509ValidationRequirement, (IX509ValidationLocation location, ConfiguredKey configuredKey)> _locationAndKeyByRequirement;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CertificateAuthenticationConfig"/> class.
+        /// </summary>
+        public CertificateAuthenticationConfig()
```

Comment:
Since it's a builder pattern this one should not be public and kick off the builder process via a Builder class

Reply:
Not sure what you mean; I think anything added above (adding yet another class) this configuration should break this "config" abstraction.

Reply:
It's already being discussed in the general thread so I'm resolving it.

Was referring to this:

```csharp
var config = CertificateAuthenticationConfigBuilder()
               .WithSubject(X509ValidationLocation.SecretProvider, subjectKey)
               .WithIssuer(X509ValidationLocation.SecretProvider, issuerKey)
               .Build();
```